app/controllers/merchants_controller.rb

```diff
@@ -0,0 +1,40 @@
+class MerchantsController < ApplicationController
+  def index
+    @merchants = Merchant.all
+  end
+
+  def show
+    @merchant = Merchant.find(params[:id])
+  end
+
+  def new
+  end
+
+  def create
+    @merchant = Merchant.create!(merchant_params)
+    redirect_to "/merchants"
+  end
+
+  def edit
+    @merchant = Merchant.find(params[:id])
+  end
+
+  def update
+    @merchant = Merchant.find(params[:id])
+    @merchant.update(merchant_params)
+    @merchant.save
+    redirect_to "/merchants/#{@merchant.id}"
+  end
+
+  def destroy
+    Item.delete(Item.where(merchant_id: params[:id]))
```

Comment:
Another way to do this is using `dependent: :destroy` in your Item model. This section of the Rails docs has more info and examples: https://guides.rubyonrails.org/association_basics.html#has-many-association-reference